tests/test_column_drag.py

```diff
@@ -48,3 +49,138 @@ def test_cd001_drag_columns(dash_duo):
     grid.wait_for_all_header_texts(["District_id", "Winner", "District"])
     grid.wait_for_pinned_cols(2)
     grid.wait_for_viewport_cols(1)
+
+def test_cd002_column_drag(dash_duo):
+    data = {
+        "ticker": ["AAPL", "MSFT", "AMZN", "GOOGL"],
+        "company": ["Apple", "Microsoft", "Amazon", "Alphabet"],
+        "price": [154.99, 268.65, 100.47, 96.75],
+        "buy": ["Buy" for _ in range(4)],
+        "sell": ["Sell" for _ in range(4)],
+        "watch": ["Watch" for _ in range(4)],
+    }
+    df = pd.DataFrame(data)
+
+    columnDefs = [
+        {
+            "headerName": "Stock Ticker",
+            "field": "ticker",
+        },
+        {"headerName": "Company", "field": "company", "filter": True},
+        {
+            "headerName": "Last Close Price",
+            "type": "rightAligned",
+            "field": "price",
+            "valueFormatter": {"function": """d3.format("($,.2f")(params.value)"""},
+        },
+    ]
+
+    defaultColDef = {
+        "resizable": True,
+        "sortable": True,
+        "editable": False,
+    }
+
+    grid = AgGrid(
+        id="topGrid",
+        columnDefs=columnDefs,
+        rowData=df.to_dict("records"),
+        columnSize="autoSize",
+        defaultColDef=defaultColDef,
+        dashGridOptions={'alignedGrids': ['middleGrid', 'bottomGrid']}
+    )
+
+    gridBot = AgGrid(
+        id="bottomGrid",
+        columnDefs=columnDefs,
+        rowData=df.to_dict("records"),
+        columnSize="autoSize",
+        defaultColDef=defaultColDef,
+    )
+
+    app = Dash(__name__)
+
+    app.layout = html.Div(
+        [
+            grid,
+            html.Div(id='hiding'),
+            gridBot,
+            html.Button(id='link')
+        ],
+        style={"margin": 20},
+    )
+
+    @app.callback(Output('hiding', 'children'), Input('link', 'n_clicks'),
+                  State('topGrid', 'columnState'))
+    def addGrid(n, s):
+        grid2 = AgGrid(
+            id="middleGrid",
+            columnDefs=columnDefs,
+            rowData=df.to_dict("records"),
+            columnSize="autoSize",
+            defaultColDef=defaultColDef,
+            dashGridOptions={'alignedGrids': 'bottomGrid'},
```

Comment:
Weird that this is necessary, but the [AG Grid docs](https://www.ag-grid.com/javascript-data-grid/aligned-grids/#event-propagation) do seem to say this is how you need to do it. In principle it should be sufficient that you've already said the top grid is aligned to both middle and bottom, so implicitly the middle and bottom are also aligned to each other, and then we should be able to let AG Grid ensure all the right events are fired without infinite loops. At any rate that's how plotly.js does the equivalent thing, when you set several axes to have matching or scaled axes... you aren't even allowed to overspecify it.